### PR TITLE
Include cert-manager Issuer + Certificate for webhook cert

### PR DESCRIPTION
**Pull request by ubergesundheit**
This PR changes the charts to include a `Issuer` and `Certificate` to make them installable out of the box. Following the instructions in the readme failed because of a missing `conversionWebhook.tlsSecretName` value.

Ping @yncyrydybyl

### CODE REVIEW

**Comment by ubergesundheit** [comment]
Changes in https://github.com/element-hq/ess-starter-edition-core/pull/8/commits/b4fc92a1dafc5eb9bee4c83ed912a5df3e219b84 are required to make the Sliding Sync proxy work.

I also found that `os.urandom` generates secrets not suitable for the synapse macaroon.

File: README.md [hunk 1]
```diff
@@ -56,25 +57,16 @@ NAME                                    URL
...
-
 #### Installing the helm charts for the `element-updater` and the `element-operator`
 
 To install the helm charts and actually deploy the `element-updater` and the `element-operator` with their default configurations, simply run:
 
 ```bash
-helm install element-updater ess-starter-edition-core/element-updater --namespace element-updater
-helm install element-operator ess-starter-edition-core/element-operator --namespace element-operator
+helm install element-updater ess-starter-edition-core/element-updater --namespace element-updater --create-namespace
+helm install element-operator ess-starter-edition-core/element-operator --namespace element-operator --create-namespace
```
Comment (hanthor) [inline README.md:66]: This is smarter/better

File: README.md [hunk 2]
```diff
@@ -87,10 +79,10 @@ element-operator-controller-manager-778c8bfbcf-4zzpl   2/2     Running   6 (8h a
 
 #### Generating the ElementDeployment CRD to Deploy Element Server Suite
 
-1) Create a CRD definition on your own starting from this base template:
+1) Create a `ElementDeployment` custom resource on your own starting from this base template:
 
    ```yaml
-   apiVersion: matrix.element.io/v1alpha1
+   apiVersion: matrix.element.io/v1alpha2
```
Comment (hanthor) [inline README.md:85]: @benbz has the apiversion incremented?

File: README.md [hunk 3]
```diff
@@ -166,35 +159,42 @@ element-operator-controller-manager-778c8bfbcf-4zzpl   2/2     Running   6 (8h a
...
+## Function to encode string as base64
+def to_base64(value):
+    return base64.b64encode(value.encode('utf-8')).decode('utf-8')
+
 ## Function to generate a secret and format it properly
-def generate_secret(name):
-    value = base64.b64encode(os.urandom(32)).decode('utf-8')
-    return f'  {name}: "{value}"'
+def generate_random(len):
+    return ''.join(random.choice(string.ascii_letters) for i in range(len))
```
Comment (hanthor) [inline README.md]: Synapse generates this using 
```
 if generate_secrets:
            macaroon_secret_key = 'macaroon_secret_key: "%s"' % (
                random_string_with_symbols(50),
            )
            form_secret = 'form_secret: "%s"' % random_string_with_symbols(50)
```

where `random_string_with_symbols` is:

```
def random_string_with_symbols(length: int) -> str:
    """Generate a cryptographically secure string of random letters/numbers/symbols.

    Drawn from the characters: `a-z`, `A-Z`, `0-9`, and `.,;:^&*-_+=#~@`
    """
    return "".join(secrets.choice(_string_with_symbols) for _ in range(length))

```

where `_string_with_symbols` is:

```
_string_with_symbols = string.digits + string.ascii_letters + ".,;:^&*-_+=#~@"

```

Comment (hanthor) [reply README.md]: Not sure if we want to add those characters into the possibilities

Comment (ubergesundheit) [reply README.md]: Nice, thanks for pointing me to this. I've updated the example script to use the additional character possibilities

**Comment by benbz** [comment]
Hi. Thank you for submitting this. We're still working out the details, but we will need a CLA signed for us to be able to incorporate any of the changes here. Myself or someone else will provide an update here when we've got the process a bit more worked out

File: README.md (hunk 2 above)
Comment (benbz) [reply README.md:85]: 2.13 incremented the version and made some small changes. The conversion webhook handles the upgrading/downgrading

**Comment by ubergesundheit** [comment]
Are you still using/supporting this operator?

**Comment by benbz** [comment]
> Are you still using/supporting this operator?

Yes. The code here is part of the operator that forms the enterprise edition of Element Server Suite, development of which happens in a private repository and the subset of that code that is included in starter edition core ends up here for each and every tag

**Comment by benbz** [comment]
Thanks for your contribution, however we now have the `ess-system` chart that wraps up the `operator`/`updater` charts and includes Cert Manager resources. We do not want to include Cert Manager resources in the `operator`/`updater` charts themselves

**Pull request closed without merging**